File: src/leetcode/python/TwoSumII_167.py
class TwoSumII_167(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        lo, hi = 0, len(numbers) - 1
        while lo < hi:
            if target > numbers[lo] + numbers[hi]:
                lo += 1
            elif target < numbers[lo] + numbers[hi]:
                hi -= 1
            else:
                break
        return [lo + 1, hi + 1]

File: src/leetcode/java/TwoSumII_167.java
package leetcode.java;

public class TwoSumII_167 {
  public int[] twoSum(int[] nums, int target) {
    int lo = 0, hi = nums.length - 1;
    while (lo < hi) {
      if (target < nums[lo] + nums[hi]) hi--;
      else if (target > nums[lo] + nums[hi]) lo++;
      else break;
    }
    return new int[] { lo + 1, hi + 1 };
  }
}
